controllers/articleController.js

```diff
@@ -165,11 +163,45 @@ const reactToArticle = async (req, res) => {
   }
 };
 
+// method for averaging all ratings of an article and
+// adding to the article object
+const addRatingsToArticle = (res, req, rated) => {
+  if (rated) {
+    return Rating.findAll({ where: { articleId: req.article.id } })
+      .then((ratings) => {
+        let ratingSum = 0;
+        ratings.forEach((rate) => { ratingSum += rate.starRating; });
+        const averageRating = ratingSum / ratings.length;
+
+        return req.article.update({ averageRating })
+          .then(ratedArticle => res.status(201).send({
+            message: `You've rated this article ${req.userRating} star`,
+            ratedArticle,
+            ratings
+          }));
+      });
+  }
+  return res.status(500).send({
+    error: 'Internal server error'
+  });
+};
+
+// method for posting the ratings of an article
+const postRating = async (req, res) => {
+  if (req.rating) {
+    const updatedRating = await helper.updateRating(req.rating, req.userRating);
+    return addRatingsToArticle(res, req, updatedRating);
+  }
+  const postedRating = await helper.addRating(req.userRating, req.userId, req.article.id);
+  return addRatingsToArticle(res, req, postedRating);
+};
+
```

Comment:
Hi Victor. Firstly I like how succinct this function is.

I have a suggestion though. I think addRatingsToArticle functionality shouldn't have 'res, req' passed into it as arguments. From experience, this might lead to errors with node where it tries to send a response twice. 

I would suggest having the function return a value which is then tested by a conditional statement, before returning a response. 

TLDR All responses should be sent from your controller in my opinion.